samples/audio-recording-add-on/src/utils/WavBlobUtil.js

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2024 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+/**
+ * A utility function to convert the raw blob coming from the media Recorder
+ * web API
+ *  from browser to a wav blob that we connect into wav blob with an
+ * HPC codec. 
+ */
+export class WavBlobUtil {
+    //Takes the rawBlob of media Recorder as input and convert it into hpc codec wav blob.
+    static async getHpcBlob(audioBlob) {
+        const url = URL.createObjectURL(audioBlob);
+        const response = await fetch(url);
+        const arrayBuffer = await response.arrayBuffer();
+        const audioContext = new AudioContext();
+        const audioBuffer = await audioContext.decodeAudioData(arrayBuffer);
+        return this.getWavBlob(audioBuffer); // Changed to use "this" to refer to the static method
+    }
+
+    // Returns Uint8Array of WAV bytes
+    static getWavBlob(audioBuffer, as32BitFloat = false) {
```

Comment:
Please make the other functions, that are not meant to be consumed by its clients, as private.